core/src/main/java/io/crate/core/collections/StringObjectMaps.java

```diff
@@ -63,4 +65,43 @@ public static Object fromMapByPath(Map value, List<String> path) {
         }
         return tmp;
     }
+
+    /**
+     * Inserts a value into source under the given key+path
+     */
+    public static void mergeInto(Map<String, Object> source, String key, List<String> path, Object value) {
+        if (path.isEmpty()) {
+            source.put(key, value);
+        } else {
+            // TODO: Make this non-recursive?
+            if (source.containsKey(key)) {
+                Map<String, Object> contents = (Map<String, Object>) source.get(key);
+                if (contents == null) {
```

Comment:
how can this happen?

Reply:
If there is an explicit `null` value in the source

```
source: {"o": null}
set o['x'] = 10
```

There is a unit test and and integration test for this case.

Reply:
right, stupid question, was thinking in the wrong direction.